ultratrace2/app.py

```diff
@@ -6,27 +7,30 @@
 
 
 class App:
-    def __init__(self, args):  # FIXME: be more granular here
+    def __init__(self, args: Namespace):
```

Comment:
I think accepting a `Namespace` here will lead to spaghetti in the future because there's no way to know what attributes exist on it. Right now it's fine because there are only two parameters we need, but as this number grows, this approach won't scale. I prefer accepting the actual parameters rather than an untyped parameter bag.
```suggestion
    def __init__(self, headless: bool = False, path: Optional[str] = None):
```

Reply:
Yeah I was worried about this.  I think you're right that unpacking the `Namespace` immediately after creation and then passing the unpacked dict to this constructor is a better idea.  Good catch.

Reply:
f7b7f7a